sf-stream/src/convert.rs

```diff
@@ -0,0 +1,481 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::protos::extractor;
+use aptos_api_types::{
+    Event, GenesisPayload, HashValue, MoveModuleId, ScriptPayload, Transaction, TransactionInfo,
+    TransactionPayload, WriteSet, WriteSetChange,
+};
+use std::str::FromStr;
+
+pub fn convert_move_module_id(move_module_id: &MoveModuleId) -> extractor::MoveModuleId {
+    extractor::MoveModuleId {
+        address: move_module_id.address.to_string(),
+        name: move_module_id.name.to_string(),
+        special_fields: Default::default(),
+    }
+}
+
+pub fn convert_transaction_payload(payload: &TransactionPayload) -> extractor::TransactionPayload {
+    match payload {
+        TransactionPayload::ScriptFunctionPayload(sfp) => extractor::TransactionPayload {
+            type_: protobuf::EnumOrUnknown::new(
+                extractor::transaction_payload::PayloadType::SCRIPT_FUNCTION_PAYLOAD,
+            ),
+            payload: Some(
+                extractor::transaction_payload::Payload::ScriptFunctionPayload(
+                    extractor::ScriptFunctionPayload {
+                        function: protobuf::MessageField::some(extractor::ScriptFunctionId {
+                            module: protobuf::MessageField::some(convert_move_module_id(
+                                &sfp.function.module,
+                            )),
+                            name: sfp.function.name.to_string(),
+                            special_fields: Default::default(),
+                        }),
+                        type_arguments: sfp
+                            .type_arguments
+                            .iter()
+                            .map(|move_type| move_type.to_string())
+                            .collect(),
+                        arguments: sfp
+                            .arguments
+                            .iter()
+                            .map(|move_value| move_value.to_string())
+                            .collect(),
+                        special_fields: Default::default(),
+                    },
+                ),
+            ),
+            special_fields: Default::default(),
+        },
+        TransactionPayload::ScriptPayload(sp) => extractor::TransactionPayload {
+            type_: protobuf::EnumOrUnknown::new(
+                extractor::transaction_payload::PayloadType::SCRIPT_PAYLOAD,
+            ),
+            payload: Some(extractor::transaction_payload::Payload::ScriptPayload(
+                extractor::ScriptPayload {
+                    code: sp.code.bytecode.to_string(),
```

Comment:
Do you really want to `to_string()` bytecode?  Don't you want to hex encode it or something?